library/oidc.c

```diff
@@ -181,9 +182,16 @@ int oidc_client_init(uv_loop_t *loop, oidc_client_t *clt,
 }
 
 int oidc_client_set_cfg(oidc_client_t *clt, const ziti_jwt_signer *cfg) {
-    clt->signer_cfg = cfg;
-    tlsuv_http_set_url(&clt->http, clt->signer_cfg->provider_url);
-    return 0;
+    free_ziti_jwt_signer(&clt->signer_cfg);
+
+    clt->signer_cfg.client_id = cfg->client_id ? strdup(cfg->client_id) : NULL;
+    clt->signer_cfg.provider_url = strdup(cfg->provider_url);
```

Comment:
all the others have a check first, should this also check? cfg->provider_url ? dup : NULL;?